src/unix/core.c

```diff
@@ -983,3 +984,61 @@ int uv__dup2_cloexec(int oldfd, int newfd) {
     return r;
   }
 }
+
+
+int uv_os_homedir(char* buffer, size_t* size) {
+  struct passwd pw;
+  struct passwd* result;
+  char* buf;
+  uid_t uid;
+  size_t bufsize;
+  size_t len;
+  int r;
+
+  if (buffer == NULL || size == NULL || *size == 0)
+    return -EINVAL;
+
+  bufsize = sysconf(_SC_GETPW_R_SIZE_MAX);
+
+  if (bufsize <= 0)
+    return -EIO;
+
+  uid = getuid();
```

Comment:
@bnoordhuis thanks for the clarification. If we're ditching thread safety, do you think it would make sense to simplify the code and replace `getpwuid_r()` with `getpwuid()`?